### ISSUE TITLE
Add rest-api resource for notification channels and subscriptions

### ISSUE THREAD

**Issue by johnhooks** [WordPress/wp-feature-notifications#179]
## What problem does this address?

The current rest-api does not support notification channels or subscriptions.

## What is your proposed solution?

I propose adding and new resources:

`/wp-json/wp/v2/notifications/channels` - Create, Read, Update, Delete notification channels.
`/wp-json/wp/v2/notifications/subscriptions` - Create, Read, Update, Delete notification subscriptions.

If we follow this pattern, the correct resource for messages would be:

`/wp-json/wp/v2/notifications/messages` not `/wp-json/wp/v2/notifications`

## Questions

There are not many nested resources in the core rest-api, is this the right approach? And there are none that use the nested url just for name spacing. The route structure that matches core better looks like :

```
/wp-json/wp/v2/notifications
/wp-json/wp/v2/channels
/wp-json/wp/v2/subscriptions
```

This then begs the question. should channels and subscriptions be pluggable beyond the scope of notifications? It could introduced an extendable messaging  system inside WordPress.

**Comment by TimothyBJacobs** [comment]
Core has reserved `wp/*` and `wp-*`. I'd recommend using `wp-notifications/v1` similar to what we did for Site Health which uses `wp-site-health/v1`.

**Comment by johnhooks** [comment]
@TimothyBJacobs Thank you, I will make that change.

**Comment by johnhooks** [comment]
@Sephsekla and @erikyo what do you think of these api routes?

```
/wp-json/wp-notifications/v1/messages
/wp-json/wp-notifications/v1/channels
/wp-json/wp-notifications/v1/subscriptions
```

**Comment by johnhooks** [comment]
Closed by #300
